Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/insightidr_test.py

```diff
@@ -0,0 +1,435 @@
+        the output of the command function with the expected output.
+    """
+    from insightidr import Client, insight_idr_download_logs_command
+
+    mock_response = util_load_file('test_data/download_logs.txt')
+    requests_mock.get(
+        f'https://{REGION}.api.insight.rapid7.com/log_search/download/logs/x:y', text=mock_response)
+
+    client = Client(
+        base_url=f'https://{REGION}.api.insight.rapid7.com/',
+        verify=False,
+        headers={
+            'Authentication': 'apikey'
+        },
+        proxy=False
+    )
+    response = insight_idr_download_logs_command(client, 'x:y')
+
+    assert (response.get('File', '')[-4:]) == '.log'
+
+
+def test_insight_idr_query_log(requests_mock) -> None:
+    """
+        Tests insight_idr_query_log_command.
+        Configures requests_mock instance to generate the appropriate
+        API response, loaded from a local JSON file. Checks
+        the output of the command function with the expected output.
+    """
+    from insightidr import Client, insight_idr_query_log_command
+
+    mock_response = util_load_json('test_data/list_log_sets.json')
+    requests_mock.get(
+        f'https://{REGION}.api.insight.rapid7.com/log_search/query/logs/x', json=mock_response)
+
+    client = Client(
+        base_url=f'https://{REGION}.api.insight.rapid7.com/',
+        verify=False,
+        headers={
+            'Authentication': 'apikey'
+        },
+        proxy=False
+    )
+    response = insight_idr_query_log_command(client, 'x', '', '', '')
+
+    outputs = []
+    for event in response.raw_response.get('events', []):
+        outputs.append(event)
+
+    assert response.outputs_prefix == 'Rapid7InsightIDR.Event'
+    assert response.outputs_key_field == 'message'
+    assert response.outputs == outputs
+
+
+def test_insight_idr_query_log_set(requests_mock) -> None:
+    """
+        Tests insight_idr_query_log_set_command.
+        Configures requests_mock instance to generate the appropriate
+        API response, loaded from a local JSON file. Checks
+        the output of the command function with the expected output.
+    """
+    from insightidr import Client, insight_idr_query_log_set_command
+
+    mock_response = util_load_json('test_data/list_log_sets.json')
+    requests_mock.get(
+        f'https://{REGION}.api.insight.rapid7.com/log_search/query/logsets/x', json=mock_response)
+
+    client = Client(
+        base_url=f'https://{REGION}.api.insight.rapid7.com/',
+        verify=False,
+        headers={
+            'Authentication': 'apikey'
+        },
+        proxy=False
+    )
+    response = insight_idr_query_log_set_command(client, 'x', '', '', '')
+
+    outputs = []
+    for event in response.raw_response.get('events', []):
+        outputs.append(event)
+
+    assert response.outputs_prefix == 'Rapid7InsightIDR.Event'
+    assert response.outputs_key_field == 'message'
+    assert response.outputs == outputs
+
+
```

Comment:
let's add more test cases for fetch incidents:
1 with a `last_run` timestamp mocked and see that we do **not** get the not needed prev incidents + last fetch is updated accordingly(+1 milisecond, right?)

1 with a `last_run` timestamp mocked and see that we **do** get the needed later incidents + last fetch is updated accordingly

Reply:
I added the check of the timestamp (and yes, it's +1 millisecond 😄) but as for the incident checking
I saw in the HelloWorld package that there is this check: 
`    assert new_incidents == [
        {
            'name': 'Hello World Alert 100',
            'occurred': '2020-02-17T23:34:23.000Z',
            'rawJSON': json.dumps(mock_response['alerts'][0]),
            'severity': 4,  # critical, this is XSOAR severity (already converted)
        },
        {
            'name': 'Hello World Alert 200',
            'occurred': '2020-02-17T23:34:23.000Z',
            'rawJSON': json.dumps(mock_response['alerts'][1]),
            'severity': 2,  # medium, this is XSOAR severity (already converted)
        }
    ]`
and I think it's better to check this than the fake API responses based on the timing (because it's fake data it does not check any functionality anyway) 
What do you think?

Reply:
it's great to have the test case which was shown in HelloWorld,. I just wanted to add more cases, on top of that.